internal/provider/resources/flow.go

```diff
@@ -0,0 +1,355 @@
+func copyFlowToModel(ctx context.Context, flow *api.Flow, model *FlowResourceModel) diag.Diagnostics {
+	model.ID = types.StringValue(flow.ID.String())
+	model.Created = customtypes.NewTimestampPointerValue(flow.Created)
+	model.Updated = customtypes.NewTimestampPointerValue(flow.Updated)
+	model.Name = types.StringValue(flow.Name)
+
+	tags, diags := types.ListValueFrom(ctx, types.StringType, flow.Tags)
+	if diags.HasError() {
+		return diags
+	}
+	model.Tags = tags
+
+	return nil
+}
+
+// Create creates the resource and sets the initial Terraform state.
+func (r *FlowResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var model FlowResourceModel
+
+	// Populate the model from resource configuration and emit diagnostics on error
+	resp.Diagnostics.Append(req.Config.Get(ctx, &model)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	var tags []string
+	resp.Diagnostics.Append(model.Tags.ElementsAs(ctx, &tags, false)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	client, err := r.client.Flows(model.AccountID.ValueUUID(), model.WorkspaceID.ValueUUID())
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Error creating flows client",
+			fmt.Sprintf("Could not create flows client, unexpected error: %s. This is a bug in the provider, please report this to the maintainers.", err.Error()),
+		)
+	}
+
+	flow, err := client.Create(ctx, api.FlowCreate{
+		Name: model.Name.ValueString(),
+		Tags: tags,
+	})
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Error creating flow",
+			fmt.Sprintf("Could not create flow, unexpected error: %s", err),
+		)
+
+		return
+	}
+
+	resp.Diagnostics.Append(copyFlowToModel(ctx, flow, &model)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	resp.Diagnostics.Append(resp.State.Set(ctx, &model)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+}
+
+// Read refreshes the Terraform state with the latest data.
+func (r *FlowResource) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	var model FlowResourceModel
+
+	// Populate the model from state and emit diagnostics on error
+	resp.Diagnostics.Append(req.State.Get(ctx, &model)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// if model.ID.IsNull() && model.Handle.IsNull() {
+	// 	resp.Diagnostics.AddError(
+	// 		"Both ID and Handle are unset",
+	// 		"This is a bug in the Terraform provider. Please report it to the maintainers.",
+	// 	)
+
+	// 	return
+	// }
+
+	client, err := r.client.Flows(model.AccountID.ValueUUID(), model.WorkspaceID.ValueUUID())
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Error creating deployment client",
+			fmt.Sprintf("Could not create deployment client, unexpected error: %s. This is a bug in the provider, please report this to the maintainers.", err.Error()),
+		)
+	}
+
+	// A deployment can be imported + read by either ID or Handle
+	// If both are set, we prefer the ID
```

Comment:
based on the logic below it looks like we're only importing by ID for now, right?

Reply:
Currently, when I try to import a flow, I get:

```
$ tf import prefect_flow.flow 85f06a57-d187-416e-b49a-8beb307b9afb
prefect_flow.flow: Importing from ID "85f06a57-d187-416e-b49a-8beb307b9afb"...
prefect_flow.flow: Import prepared!
  Prepared prefect_flow for import
prefect_flow.flow: Refreshing state...
╷
│ Error: Plugin did not respond
│
│ The plugin encountered an error, and failed to respond to the plugin6.(*GRPCProvider).ReadResource call. The plugin logs may contain more details.
╵


Stack trace from the terraform-provider-prefect plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x0 pc=0x105491a50]

goroutine 66 [running]:
github.com/prefecthq/terraform-provider-prefect/internal/provider/resources.copyFlowToModel({0x1057aa0e8, 0x14000489560}, 0x0, 0x140000ee3c0)
        /Users/mitch/code/github.com/prefecthq/terraform-provider-prefect/resource_flow/internal/provider/resources/flow.go:141 +0x30
github.com/prefecthq/terraform-provider-prefect/internal/provider/resources.(*FlowResource).Read(0x1400048ae50, {0x1057aa0e8, 0x14000489560}, {{{{0x1057af758, 0x140005160f0}, {0x10569c100, 0x14000516030}}, {0x1057b11d8, 0x14000433040}}, 0x140004440c0, ...}, ...)
        /Users/mitch/code/github.com/prefecthq/terraform-provider-prefect/resource_flow/internal/provider/resources/flow.go:258 +0x39c
github.com/hashicorp/terraform-plugin-framework/internal/fwserver.(*Server).ReadResource(0x1400018d340, {0x1057aa0e8, 0x14000489560}, 0x140004895f0, 0x14000297690)
        /Users/mitch/go/pkg/mod/github.com/hashicorp/terraform-plugin-framework@v1.4.2/internal/fwserver/server_readresource.go:101 +0x510
github.com/hashicorp/terraform-plugin-framework/internal/proto6server.(*Server).ReadResource(0x1400018d340, {0x1057aa0e8?, 0x14000488e70?}, 0x140004a2600)
        /Users/mitch/go/pkg/mod/github.com/hashicorp/terraform-plugin-framework@v1.4.2/internal/proto6server/server_readresource.go:55 +0x2e0
github.com/hashicorp/terraform-plugin-go/tfprotov6/tf6server.(*server).ReadResource(0x1400038e000, {0x1057aa0e8?, 0x140004884e0?}, 0x14000530060)
        /Users/mitch/go/pkg/mod/github.com/hashicorp/terraform-plugin-go@v0.22.1/tfprotov6/tf6server/server.go:776 +0x3c8
github.com/hashicorp/terraform-plugin-go/tfprotov6/internal/tfplugin6._Provider_ReadResource_Handler({0x105778520, 0x1400038e000}, {0x1057aa0e8, 0x140004884e0}, 0x140001a4800, 0x0)
        /Users/mitch/go/pkg/mod/github.com/hashicorp/terraform-plugin-go@v0.22.1/tfprotov6/internal/tfplugin6/tfplugin6_grpc.pb.go:482 +0x1c0
google.golang.org/grpc.(*Server).processUnaryRPC(0x140001a1000, {0x1057aa0e8, 0x140004883f0}, {0x1057b0060, 0x140000bf860}, 0x1400021e000, 0x14000383050, 0x105d26688, 0x0)
        /Users/mitch/go/pkg/mod/google.golang.org/grpc@v1.62.1/server.go:1386 +0xb58
google.golang.org/grpc.(*Server).handleStream(0x140001a1000, {0x1057b0060, 0x140000bf860}, 0x1400021e000)
        /Users/mitch/go/pkg/mod/google.golang.org/grpc@v1.62.1/server.go:1797 +0xb10
google.golang.org/grpc.(*Server).serveStreams.func2.1()
        /Users/mitch/go/pkg/mod/google.golang.org/grpc@v1.62.1/server.go:1027 +0x8c
created by google.golang.org/grpc.(*Server).serveStreams.func2 in goroutine 13
        /Users/mitch/go/pkg/mod/google.golang.org/grpc@v1.62.1/server.go:1038 +0x13c

Error: The terraform-provider-prefect plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```

Reply:
hmm that's odd. the acceptance test for import passes fine. Let me take a closer look at this one.

Reply:
ok, some investigation into this. Import is working for the following scenario:
`terraform import prefect_flow.flow workspace_id,flow_id`
But in other scenarios it panics.
I'm looking into it.

Reply:
ok, I made a couple of updates to support import correctly using ids. 
It's possible to do this via name too. 
But suggest we wait to see if any users request this as it meets the basic requirements currently.
Please take another look.

Reply:
Really appreciate the iteration here, thanks @skyscrapr.

Looks like we just need to better handle the scenario where the `inputParts` only equals `1`. You can see [block's import logic as an example](https://github.com/PrefectHQ/terraform-provider-prefect/blob/5ff52364630ea269fc336c8e66601a3cb381ad26/internal/provider/resources/block.go). I was able to copy some of that logic and get `tf import` working:

```diff
diff --git a/internal/provider/resources/flow.go b/internal/provider/resources/flow.go
index d26e379..97c609d 100644
--- a/internal/provider/resources/flow.go
+++ b/internal/provider/resources/flow.go
@@ -333,10 +333,17 @@ func (r *FlowResource) ImportState(ctx context.Context, req resource.ImportState
 		return
 	}
 
-	id := inputParts[maxInputCount-1]
-	if len(inputParts) == maxInputCount {
-		workspaceID := inputParts[0]
-		resp.Diagnostics.Append(resp.State.SetAttribute(ctx, path.Root("workspace_id"), workspaceID)...)
+	identifier := inputParts[0]
+	resp.Diagnostics.Append(resp.State.SetAttribute(ctx, path.Root("id"), identifier)...)
+
+	if len(inputParts) == 2 && inputParts[1] != "" {
+		workspaceID, err := uuid.Parse(inputParts[1])
+		if err != nil {
+			resp.Diagnostics.AddError("problem parsing workspace ID", "see details")
+
+			return
+		}
+
+		resp.Diagnostics.Append(resp.State.SetAttribute(ctx, path.Root("workspace_id"), workspaceID.String())...)
 	}
-	resp.Diagnostics.Append(resp.State.SetAttribute(ctx, path.Root("id"), id)...)
 }
```

```shell
$ tf import prefect_flow.flow c9bb05e4-2efe-49f7-b4e2-318d295553b6
prefect_flow.flow: Importing from ID "c9bb05e4-2efe-49f7-b4e2-318d295553b6"...
prefect_flow.flow: Import prepared!
  Prepared prefect_flow for import
prefect_flow.flow: Refreshing state... [id=c9bb05e4-2efe-49f7-b4e2-318d295553b6]

Import successful!

The resources that were imported are shown above. These resources are now in
your Terraform state and will henceforth be managed by Terraform.
```

>  But suggest we wait to see if any users request this as it meets the basic requirements currently.

I'm not opposed to that, but let's at least confirm that diagnostics properly return this information to communicate that workspace ID currently needs to be supplied (meaning this won't work on OSS for now). Currently, `id := inputParts[maxInputCount-1]` throws an error via stack trace rather than diagnostics. So let's either get that method working based of my example above, or throw correct diagnostics for now 👍🏼

Reply:
ok, applied suggested changes.